### ISSUE TITLE
Omitted Declaration

### ISSUE THREAD

**Issue by mzovadia** [jakubpawlowicz/clean-css#615]
Not sure if this is an issue or an option setting I'm not aware of. In the following rule, for example,

.rule {
padding: 1.56em;
padding-bottom: .56em;
}

The minified result omits the first padding declaration, leaving only "padding-bottom: .56em". Is there a way to make the minified version include both padding declarations? Thanks.

**Comment by mzovadia** [comment]
OK, so I figured out that setting the option advanced:false solves the issue, but now I don't see any css rules when I try to inspect an element in Chrome's dev tools.

**Comment by mzovadia** [comment]
Turned out to be a Chrome thing. Closing . . .

**Comment by jakubpawlowicz** [comment]
:+1: it definitely works with both advanced turned on and off - just tested on the latest 3.3.

**Comment by mzovadia** [comment]
Oh, OK. I should have been a little more explicit in detailing the how the rule is written in the Sass file, as it appears to make a difference:

.rule {
  padding: $general-component-padding;
  padding-bottom: $general-component-padding - 1em;
}

So, with advanced turned on, the first padding declaration is omitted from the minified file. Both declarations are present when advanced is turned off.

**Comment by jakubpawlowicz** [comment]
Please correct me if I'm wrong but Sass is a pre-processor so at the minification step both of these should have numeric values rather than variables, right?

**Comment by mzovadia** [comment]
Yes, the variables have numeric values at the minification step. Thus, the unminified CSS file contains:

.rule {
padding: 1.56em;
padding-bottom: .56em;
}

The result following minification of this file with advanced turned off is:

.rule {padding:1.56em;padding-bottom:.56em}

With advanced turned on, it is:

.rule {padding-bottom:.56em}

**Comment by jakubpawlowicz** [comment]
Can you find out which version of clean-css do you use with `npm ls`?

**Comment by mzovadia** [comment]
Yup, I'm using 3.3.5

OK, I think I've tracked down the issue. I have a clearfix mixin, which I omitted from the example in order to keep things tidy. it looks like this:

@mixin clearfix {
  &:before, &:after {
    content: " ";
    display: table;
  }
  &:after {
    clear: both;
  }
  zoom: 1;
  /* for IE 7 */
}

So, this is what the whole rule looks like, pre-minification, with the clearfix:

.rule {
  zoom: 1;
  /* for IE 7 */
  border-top: 1px solid #dddcdc;
  background: #f7f7f7;
  padding: 1.56em;
  padding-bottom: 0.56em;
  margin-bottom: 2em; 
}
  .rule:before,
  .rule:after {
    content: " ";
    display: table; 
}
  .rule:after {
    clear: both; 
}


This is what it looks like post-minification: 

.rule{zoom:1;border-top:1px solid #dddcdc;background:#f7f7f7;padding-bottom:.56em;margin-bottom:2em}

This is what it looks like post-minification with advanced set to false: 

.rule{zoom:1;border-top:1px solid #dddcdc;background:#f7f7f7;padding:1.56em;padding-bottom:.56em;margin-bottom:2em}.rule:after,.rule:before{content:" ";display:table}.rule:after{clear:both}.

To sum up:

When minified with clearfix included, the first padding declaration and clearfix declarations are missing (except zoom:1) 

When advanced is set to false, all works as expected with clearfix.

When the clearfix mixin is removed, all else works as expected.

**Comment by jakubpawlowicz** [comment]
Thanks for coming back, I'll give it another look.

**Comment by mzovadia** [comment]
Thanks, Jakub. Have you had a chance to look at this? Please let me know if you need clarification on anything.

**Comment by jakubpawlowicz** [comment]
Not yet as I was out for a while. But I will.

**Comment by jakubpawlowicz** [comment]
I have retested it in the latest 3.3 and master (pre 3.4) and it works fine now.

Advanced on:
```css
.rule{zoom:1;border-top:1px solid #dddcdc;background:#f7f7f7;padding:1.56em 1.56em .56em;margin-bottom:2em}.rule:after,.rule:before{content:" ";display:table}.rule:after{clear:both}
```

Advanced off:
```css
.rule{zoom:1;border-top:1px solid #dddcdc;background:#f7f7f7;padding:1.56em;padding-bottom:.56em;margin-bottom:2em}.rule:after,.rule:before{content:" ";display:table}.rule:after{clear:both}
```

Sorry you had to wait a bit.
